cl/sentinel/gossip.go

```diff
@@ -651,8 +651,22 @@ func (s *GossipSubscription) run(ctx context.Context, sub *pubsub.Subscription,
 }
 
 func (g *GossipSubscription) Publish(data []byte) error {
-	if len(g.topic.ListPeers()) == 0 {
-		log.Debug("[Gossip] No peers to publish to for topic", "topic", g.topic.String())
+	if len(g.topic.ListPeers()) < 2 {
+		log.Trace("[Gossip] No peers to publish to for topic", "topic", g.topic.String())
+		go func() {
```

Comment:
So this is to avoid the whole routine getting blocked during the underlying peer search mechanism?

Reply:
yes